sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/RequestResponseChannel.java

```diff
@@ -201,15 +204,33 @@ protected RequestResponseChannel(AmqpConnection amqpConnection, String connectio
 
     @Override
     public Mono<Void> closeAsync() {
+        final Mono<Void> closeOperationWithTimeout = closeMono.asMono()
+            .timeout(retryOptions.getTryTimeout())
+            .onErrorResume(TimeoutException.class, error -> {
+                return Mono.fromRunnable(() -> {
```

Comment:
Do you need this `onErrorResume` to emit anything other than `onComplete`? If so, you'll want to use `Mono.create` or `Mono.fromCallable` instead as `Mono.fromRunnable` will only ever emit `onComplete`.

Reply:
```java
.onErrorResume(TimeoutException.class, error -> {
    return Mono.create(sink -> {
        sink.onRequest(request -> {
            logger.info("connectionId[{}] linkName[{}] Timed out waiting for RequestResponseChannel to complete"
                    + "closing. Manually closing.",
                connectionId, linkName, error);

            onTerminalState("SendLinkHandler");
            onTerminalState("ReceiveLinkHandler");

            sink.success();
        });
    });
})
```